web/src/cypress/integration/schedules/temporarySchedule.ts

```diff
@@ -0,0 +1,284 @@
+import { Chance } from 'chance'
+import { testScreen } from '../../support'
+import { Schedule, User } from '../../../schema'
+import { DateTime, Interval } from 'luxon'
+import { round } from 'lodash-es'
+
+const c = new Chance()
+const dtFmt = "yyyy-MM-dd'T'HH:mm"
+
+function makeIntervalDates(): [string, string, number] {
+  const now = DateTime.local()
```

Comment:
I'll see how I can make this a little more robust (setting tz to current filter toggle after creating), but do you have any examples of where/how it would fail when we get to DST?

Reply:
We can take a look once the random bits are in, but basically, anywhere you pull local time, and then add & assert it may not be what is expected.

I saw that below year, month, day is being set, and then subtracting minutes -- it's a lot to follow with regard to e.g. what about 1 minute before or after DST, that type of thing.

Finding specific examples can be tricky, it's easier to make sure we're always operating in a known state. Ideally, we'd be able to set the system clock & timezone.

---

Is the purpose of this function to generate a random interval? If so maybe just use `DateTime.fromDate(Chance.date)` and add a duration to the start. Then we sidestep things like DST, leap year, etc..

We could also maybe return the `luxon.Interval` too which has start, end, and duration bits. That way we can stick with the luxon types until we the very moment we need to convert them to the almost-ISO string for the form fields.

Reply:
Yeah that is the purpose, though there are some parameters we have to meet such as ensuring the start date is after `now()`, and that the intervals never exceed each other (i.e. start after end, end before start, end before now, etc). I've been working through all the different cases to ensure we don't get a 422 back on the create but I'm still working through it — I've pushed what I have to origin though if you want to take a look

Reply:
I simplified it mimic what I have in the support create func, using chance and intervals